src/framework/ui/iuiconfiguration.h

```diff
@@ -49,15 +47,21 @@ class IUiConfiguration : MODULE_EXPORT_INTERFACE
     virtual QStringList possibleFontFamilies() const = 0;
     virtual QStringList possibleAccentColors() const = 0;
 
+    virtual bool isDarkMode() const = 0;
+    virtual void setIsDarkMode(bool dark, bool overrideFollowSystemTheme) = 0;
+
     virtual bool isHighContrast() const = 0;
     virtual void setIsHighContrast(bool highContrast) = 0;
 
-    virtual void resetCurrentThemeToDefault(const ThemeCode& codeKey) = 0;
-
     virtual const ThemeInfo& currentTheme() const = 0;
-    virtual void setCurrentTheme(const ThemeCode& codeKey) = 0;
-    virtual void setCurrentThemeStyleValue(ThemeStyleKey key, const Val& val) = 0;
     virtual async::Notification currentThemeChanged() const = 0;
+    virtual void setCurrentTheme(const ThemeCode& codeKey, bool overrideFollowSystemTheme) = 0;
```

Comment:
I'm not sure it is a good idea to expose "overrideFollowSystemTheme" in this method. Changing the current theme and enabling the "follow system theme" option are conflicting operations

Reply:
I went for a slightly different solution (namely splitting it into "setCurrentTheme" (public) and "doSetCurrentTheme" (private).